### ISSUE TITLE
t2 root

### ISSUE THREAD

**Issue by Frijol** [tessel/t2-cli#80]
Current status: no such command

Spec: [T2 CLI spec](https://forums.tessel.io/t/t2-cli-spec-comments-requested/1743)
**tessel root [-i <path>]**

Provides access to the underlying Linux system running on Tessel 2: exactly SSH (copies regular SSH keys to authorize and SSHes in as Linux computer)

```sh
$ tessel root --help
> Usage: tessel root [-i <path>] [--help]
>  -i <path>:	provide a path to the desired ssh key
$ 

$ tessel root
> Accessing root...
root@192.168.128.124 # 
```

**Comment by nplus11** [comment]
I recommend that `-i` option be called `-k|--keypath` in order to stay consistent with our new `tessel remote` command.

**Comment by johnnyman727** [comment]
Actually, we should probably just change the other API since `-i` is the standard. Thanks for catching that!

**Comment by Student007** [comment]
@johnnyman727 assign it to me - I have figured out this successfully. Next days I will do a pull requst after I wrote tests for MacOSX, Windows, Linux behaviors and cleaning up my code.

**Comment by Student007** [comment]
@johnnyman727 I haven't done writing test now. But cleaning up code is done. So I made a brunch "t2 root" you could take a preview. It isn't tested on Windows or Linux - only MacOSX. I have to stop for today. Tomorrow I plan to start writing tests ... but family ... weekend .. surprisings ...

**Comment by johnnyman727** [comment]
@Student007 It's a bit hard to look over the changes on your fork b/c it's spread across to many different repos. Could you squash your commits and open a PR on this repo? I'm happy to review it!

**Comment by Student007** [comment]
@johnnyman727 yes of cause - when tests are written. I had gone on vacation last few days. The note about the branch was meant about to have a first look how I solved the task. So you could say "no ... not that way" or ... "fitting solution". `git clone https://github.com/Student007/t2-cli/tree/t2-root` ... than test it with several numbers of (not) authorized LAN Tessels.
If only one Tessel is out there, it connects directly - if there found more Tessels it provides a Menu-Navigation (ncurses like). It could be more improved by a much more powerful JavaScript reimplementation of ncurses (e.g.) and also additional information like CPU/Memory usage of each found Tessel,  but I like to finish testing before. I have only used Mocha and my own Test-Framework before and need some days to figure out `Sinon` and the right usage of `Mocks`, `Stubs` and `Spies`. I am also looking into the other tests to do it similar like you have done - but the controller.root I will put into a separate test file (could be moved later).

Of cause I will squash the commits like @LinusU taught me
:mortar_board: 
:octocat: 
... give me some days - then I will make a pull request after squashing commits.

**Comment by Student007** [comment]
DONE :smile: 
https://github.com/tessel/t2-cli/pull/290

**Comment by Frijol** [comment]
Updated to https://github.com/tessel/t2-cli/pull/307

**Comment by johnnyman727** [comment]
Closed with #453.
